src/tiled/stampbrush.cpp

```diff
@@ -386,6 +401,52 @@ void StampBrush::drawPreviewLayer(const QVector<QPoint> &list)
             mapDocument()->unifyTilesets(variation.map, mMissingTilesets);
 
             TileLayer *stamp = variation.tileLayer();
+            bool copyFlag = false;
+
+            //if staggered map, makes sure stamp stays the same
+            if ((mapDocument()->map()->orientation() == Map::Hexagonal
+                    || mapDocument()->map()->orientation() == Map::Staggered)
+                    && ((mapDocument()->map()->staggerAxis() == Map::StaggerY)?
+                         stamp->height() > 1 : stamp->width() > 1)) {
+                stamp = variation.tileLayer()->copy(variation.tileLayer()->bounds());
+                copyFlag = true;
+
+                if(mapDocument()->map()->staggerAxis() == Map::StaggerY) {
+                    int rowOfTop = p.y() - stamp->height() / 2;
+
+                    if ((variation.map->staggerIndex() == mapDocument()->map()->staggerIndex()
+                            && (rowOfTop&1) == 1)
```

Comment:
Coding style: please put spaces around `&`.

Also, since you're not using `rowOfTop` for any other computation, you can change that local variable to `bool oddOffset` for example.